src/stuf/stuf_bg/tests/test_client.py

```diff
@@ -142,71 +114,97 @@ def test_order_of_attributes_in_template_is_correct(self):
             self.client.get_values_for_attributes(test_bsn, all_available_attributes)
 
         request_body = m.last_request.body
-        doc = etree.parse(BytesIO(request_body))
-        soap_body = (
-            doc.getroot()
-            .xpath(
-                "soap:Body",
-                namespaces={"soap": "http://schemas.xmlsoap.org/soap/envelope/"},
-            )[0]
-            .getchildren()[0]
-        )
+        soap_body = _extract_soap_body(request_body)
 
         # validate soap message with xsd
         # order of all elements in request doc does matter!
         xmlschema.assert_(soap_body)
 
+    def test_soap_request_method_and_headers(self):
+        test_bsn = "999992314"
+        with requests_mock.Mocker() as m:
+            m.post(self.client.service.soap_service.url)
+            # perform request
+            self.client.get_values_for_attributes(test_bsn, FieldChoices.values)
+
+        self.assertEqual(m.last_request.method, "POST")
+        self.assertEqual(
+            m.last_request.headers["Content-Type"],
+            SOAP_VERSION_CONTENT_TYPES.get(SOAPVersion.soap11),
+        )
+        self.assertEqual(
+            m.last_request.headers["SOAPAction"],
+            "http://www.egem.nl/StUF/sector/bg/0310/npsLv01",
+        )
+
+    def test_soap_request_gets_logged(self):
+        test_bsn = "999992314"
+        with requests_mock.Mocker() as m:
+            m.post(self.client.service.soap_service.url)
+            # perform request
+            self.client.get_values_for_attributes(test_bsn, FieldChoices.values)
+
+        self.assertEqual(
+            TimelineLogProxy.objects.filter(
+                template="logging/events/stuf_bg_request.txt"
+            ).count(),
+            1,
+        )
```

Comment:
perhaps we should make a mixin/helper for this in `openforms.utils.tests.logging`, but I've been writing these a bit different lately:

```py
log_events = {log.event for log in TimelineLogProxy.objects.all()}
self.assertIn("stuf_bg_request", log_events)
```
In many of these assertions the exact count doesn't really matter too much, but you could easily turns this into a list comprehension with count checks too. Some helpers like `self.assertLogEvent("stuf_bg_request", count=None)` could be useful?

Reply:
Yes or something with a similar API to TestCase.assertLogs
```
with self.assertLogsToTimeLine(tag=..., event=..., count=...):
   ...
```

Reply:
Or maybe the count is implied, and the context manager records the count on `__enter__` and asserts it is greater on `__exit__`.

Reply:
Let's let this one simmer a bit. I've had a quick glance of the assertions made across tests, but I'm still a bit hazy about how timelinelogger does its thing. I don't want to slow down our suite with queries for assertions that belong in the library. DontTestThePlatform™

Reply:
sure - simmering is good

w/r to the implied count - not necessarily, e.g. the registration task retry machinery would explicitly test for multiple log entries since every retry needs to log its attempt